### ISSUE TITLE
Incorrect LR scaling in POYO and POYO+ example

### ISSUE THREAD

**Issue by divyansha1115** [neuro-galaxy/torch_brain#162]
In the POYO / POYO+ example training script, the learning rate is currently scaled using the **per-GPU batch size** only. This is incorrect for multi-GPU training because the **effective global batch size increases with `world_size`**, so LR scaling should incorporate that.

Currently it effectively does:
https://github.com/neuro-galaxy/torch_brain/blob/582534fa207dfa7c3c75f6c221b207373ea8f00d/examples/poyo/train.py#L54

However, for distributed training the effective batch size is:

```py
global_batch_size = world_size * per_gpu_batch_size
```

As a result, the LR ends up too small for multi-GPU runs, and training behavior changes depending on GPU count.

### Proposed Fix: scale LR by effective global batch size

```py
max_lr = self.cfg.optim.base_lr * self.trainer.world_size * self.cfg.batch_size
```

Optionally (and arguably preferable), apply sqrt scaling:
```py
max_lr = self.cfg.optim.base_lr * np.sqrt(self.trainer.world_size * self.cfg.batch_size)
```

Suggested improvement: It might be better if the config/CLI argument batch_size refers to the global batch size, and the per-GPU batch size is derived internally when constructing the dataloader.

```py
batch_size = self.cfg.batch_size // self.trainer.world_size
```
